src/chess/Game.js

```diff
@@ -8,5 +8,307 @@ export class Game {
+    this.renderCountMoves(this.allMovesOfStart);
+  }
+
+  // ui
+  setGamerName({ name, color }) {
+    const span = document.querySelector(`#${color}GamerName`);
+    span.textContent = name;
+  }
+
+  changeWhooseMove(color) {
+    this.whooseMove = COLORS.w === color ? COLORS.b : COLORS.w;
+  }
+
+  addToHistory(figure, from, to) {
+    const obj = { name: figure, from: from, to: to }
+    this.historyMoves.push(obj);
+
+    this.renderHistory(obj);
+  }
+
+  isPawnTransform(figure) {
+    return figure.name === FIGURE_NAMES.pawn &&
+      (figure.position[1] === 7 || figure.position[1] === 0);
+  }
+
+  moveFigure(figure) {
+    console.log(figure.position, figure.isActive);
+    if (figure.color !== this.whooseMove) return;
+    if (!!this.allowedMoves.length && figure.isActive) {
+      this.removeAllowedMoves();
+      figure.isActive = false;
+      return;
+    }
+    if (!!this.allowedMoves.length && !figure.isActive) {
+      const activeFigure = this.board.getFigureIsActive(true);
+      activeFigure.isActive = false;
+      this.removeAllowedMoves();
+    }
+
+    const moves = figure.getAllowedMoves(this.board.allFields);
+    this.renderMoves(figure, moves, (position) => {
+      this.#changeCountMoves();
+      const name = `${figure.color}-${figure.name}`;
+
+      this.destroyFigure(position);
+
+      this.board.updateAllFields(name, figure.position, position);
+      this.addToHistory(name, figure.position, position);
+      this.changeWhooseMove(figure.color);
+
+      const x = this.arrayTransforms[position[1]][position[0]][0];
+      const y = this.arrayTransforms[position[1]][position[0]][1];
+      figure.uiElement.style.transform = `translate(${x}px, ${y}px)`;
+
+      figure.position = position;
+      if (figure.name === FIGURE_NAMES.pawn) {
+        figure.isFirstStep = false;
+      }
+      figure.isActive = false;
+
+      // pawn -> new figure
+      if (this.isPawnTransform(figure)) {
+        this.modalNewFigure(figure, this.changePawnToNewFigure);
+        figure.uiElement.remove();
+      }
+
+      this.removeAllowedMoves();
+      this.nextStep();
+    });
+  }
+
+  changePawnToNewFigure = (figure, name) => {
+    const newFigure = this.board.changePawn(name, figure);
+    const [x, y] = newFigure.position;
+    const xt = this.arrayTransforms[y][x][0];
+    const yt = this.arrayTransforms[y][x][1];
+    newFigure.uiElement.style.transform = `translate(${xt}px, ${yt}px)`;
+    newFigure.uiElement.onclick = () => this.moveFigure(newFigure);
+  }
+
+  destroyFigure(position) {
+    const figure = this.board.getFigureByCoords(position);
+    if (figure) {
+      this.board.destroyFigure(position);
+      figure.uiElement.remove();
+    }
+  }
+
+  // ui
+  renderHistory({ name, from, to }) {
+    const move = document.createElement('div');
+    const [color, nameText] = name.split('-');
+    const posFrom = COORDINATES[from[1]][from[0]];
+    const posTo = COORDINATES[to[1]][to[0]];
```

Comment:
Только тут заметил, что координаты хранятся как то странно =)
Почему не
```suggestion
    const posFrom = COORDINATES[from[0]][from[1]];
    const posTo = COORDINATES[to[0]][to[1]];
```

Reply:
Их переварачивать надо, так как многомерный массив и с предложенным вариантом мы возьмомем не ту ячейку.